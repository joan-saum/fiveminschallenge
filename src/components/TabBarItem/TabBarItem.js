// @flow

import React, {useCallback} from 'react';
import {TouchableOpacity, View, Text} from 'react-native';

import type {Node} from 'react';

import styles from './TabBarItem.style';

type Props = {|
  title: string,
  onPress: (title: string) => void,
|};

const TabBarItem = ({title, onPress}: Props): Node => {
  const onNavPress = useCallback(() => {
    onPress(title);
  }, [onPress, title]);

  return (
    <TouchableOpacity onPress={onNavPress}>
      <View style={styles.container}>
        <Text style={styles.text}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default TabBarItem;
