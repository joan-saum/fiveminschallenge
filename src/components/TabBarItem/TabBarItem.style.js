import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  container: {
    height: 64,
    justifyContent: 'center',
  },
  text: {
    color: 'white',
  },
});

export default style;
