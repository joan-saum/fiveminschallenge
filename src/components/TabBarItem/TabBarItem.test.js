// @flow

import * as React from 'react';
import renderer from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';

import TabBarItem from './TabBarItem';

jest.unmock('./TabBarItem');

describe('TabBarItem', () => {
  it('should render a TabBarItem', () => {
    const onPress = jest.fn().mockName('onPress');
    const tree = renderer.create(
      <TabBarItem onPress={onPress} title="title" />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should trigger the onPress callback when the component is clicked', () => {
    const onPress = jest.fn().mockName('onPress');
    const title = 'title';
    const tree = renderer.create(
      <TabBarItem onPress={onPress} title={title} />,
    );

    expect(onPress).toHaveBeenCalledTimes(0);

    tree.root.findByType(TouchableOpacity).props.onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
    expect(onPress).toHaveBeenCalledWith(title);
  });
});
