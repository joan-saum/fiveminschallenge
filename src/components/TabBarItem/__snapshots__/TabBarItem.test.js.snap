// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarItem should render a TabBarItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 64,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
